lib/floe/workflow/intrinsic_function/transformer.rb

```diff
@@ -50,7 +55,8 @@ def check_types(args, function, signature)
             args.zip(signature).each_with_index do |(arg, type), index|
               type = type.type if type.kind_of?(OptionalArg)
 
-              raise ArgumentError, "wrong type for argument #{index + 1} to #{function} (given #{arg.class}, expected #{type})" unless arg.kind_of?(type)
+              matched = type == Boolean ? type.kind_of?(arg) : arg.kind_of?(type)
```

Comment:
I wonder if we can do this better with something like `===`, which can handle kind_of logic?

Reply:
ooh. we are not asking if type is a kind of Boolean, we are asking if type == Boolean.
Introducing `===` makes the tests fail